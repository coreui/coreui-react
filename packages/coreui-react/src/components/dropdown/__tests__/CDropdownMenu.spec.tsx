import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { CDropdown, CDropdownMenu } from '../index'

test('loads and displays CDropdownMenu component', async () => {
  const { container } = render(<CDropdownMenu>Test</CDropdownMenu>)
  expect(container).toMatchSnapshot()
})

test('CDropdownMenu customize', async () => {
  const { container } = render(
    <CDropdown visible={true}>
      <CDropdownMenu className="bazinga" as="div">
        Test
      </CDropdownMenu>
    </CDropdown>,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild?.firstChild).toHaveClass('bazinga')
  expect(container.firstChild?.firstChild).toHaveClass('dropdown-menu')
  expect(container.firstChild?.firstChild).toHaveClass('show')
})
