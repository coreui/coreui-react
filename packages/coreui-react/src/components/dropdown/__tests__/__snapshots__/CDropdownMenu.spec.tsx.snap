// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CDropdownMenu customize 1`] = `
<div>
  <div
    class="btn-group"
  >
    <div
      class="dropdown-menu show bazinga"
      role="menu"
    >
      Test
    </div>
  </div>
</div>
`;

exports[`loads and displays CDropdownMenu component 1`] = `
<div>
  <ul
    class="dropdown-menu"
    data-coreui-popper="static"
    role="menu"
  />
</div>
`;
